### PR TITLE
group device specific properties

### PR DESCRIPTION
**Pull request by rjodinchr**

### CODE REVIEW

File: src/device_properties.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include "cl_headers.hpp"
+
+struct cvk_device_properties {
+    virtual cl_ulong get_global_mem_cache_size() const { return 0; }
+    virtual cl_ulong get_num_compute_units() const { return 1; }
+
+    virtual cl_uint get_max_cmd_batch_size() const { return 10000; }
+    virtual cl_uint get_max_first_cmd_batch_size() const { return 10000; }
+    virtual cl_uint get_max_cmd_group_size() const { return UINT32_MAX; }
+    virtual cl_uint get_max_first_cmd_group_size() const { return UINT32_MAX; }
```
Comment (kpet) [inline src/device_properties.hpp:29]: Please use the default values from the config so we don't have default values in two places.

**Review by kpet — CHANGES_REQUESTED** [review]
I was thinking I'd extend the config management to do this, possibly by adding support for config sections and/or profiles, but this approach works too and is definitely a step in the right direction. We can always rework it later, which will have been made much easier by the abstraction you're introducing. Please fix the source of the default values and it's good to go. Thanks!

File: src/device_properties.hpp (hunk 1 above)
Comment (rjodinchr) [reply src/device_properties.hpp]: Done

**Review by kpet — APPROVED** [review]
Thanks!

**Pull request merged**